pkg/providers/property/data_akamai_search.go

```diff
@@ -0,0 +1,115 @@
+package property
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/akamai/AkamaiOPEN-edgegrid-golang/v2/pkg/papi"
+	"github.com/akamai/terraform-provider-akamai/v2/pkg/akamai"
+	"github.com/akamai/terraform-provider-akamai/v2/pkg/tools"
+)
+
+func dataSourceSearch() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceSearchRead,
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "Key must have one of three values: \"edgeHostname\", \"hostname\" or \"propertyName\"",
+				ValidateDiagFunc: tools.IsNotBlank,
```

Comment:
Please, validate in schema as well that the key is one of those 3 mentioned above :-)

Reply:
changed and moved to pkg/providers/property/data_akamai_property_search.go